Upstream-Status: Pending

Using uname like this when cross compiling is a really bad idea. We
provide the correct linker flags, lets just assume we can get this
right ourselves.

RP 2012/04/26
---
 bin/mklib |   16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/bin/mklib b/bin/mklib
index b33aa90..16cb017 100755
--- a/bin/mklib
+++ b/bin/mklib
@@ -330,15 +330,7 @@ case $ARCH in
 	    ;;
 	    esac
 
-	    # Check if objects are 32-bit and we're running in 64-bit
-	    # environment.  If so, pass -m32 flag to linker.
 	    set ${OBJECTS}
-	    ABI32=`file $1 | grep 32-bit`
-	    ARM=`file $1 | grep ARM`
-	    # Do not add "-m32" option for arm.
-            if [ -z "$ARM" -a "${ABI32}" -a `uname -m` = "x86_64" ] ; then
-		OPTS="-m32 ${OPTS}"
-	    fi
 
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
@@ -389,15 +381,7 @@ case $ARCH in
 		# exptmp is removed below
 	    fi
 
-	    # Check if objects are 32-bit and we're running in 64-bit
-	    # environment.  If so, pass -m32 flag to linker.
 	    set ${OBJECTS}
-	    ABI32=`file $1 | grep 32-bit`
-            ARM=`file $1 | grep ARM`
-	    # Do not add "-m32" option for arm.
-            if [ -z "$ARM" -a "${ABI32}" -a `uname -m` = "x86_64" ] ; then
-		OPTS="-m32 ${OPTS}"
-	    fi
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
             fi
-- 
1.7.10.4

